src/osvr/Common/EyeTrackerComponent.cpp

```diff
@@ -0,0 +1,150 @@
+/** @file
+    @brief Implementation
+
+    @date 2015
+
+    @author
+    Sensics, Inc.
+    <http://sensics.com/osvr>
+
+*/
+
+// Copyright 2015 Sensics, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Internal Includes
+#include <osvr/Common/EyeTrackerComponent.h>
+#include <osvr/Common/BaseDevice.h>
+#include <osvr/Common/Serialization.h>
+#include <osvr/Common/Buffer.h>
+#include <osvr/Util/Verbosity.h>
+
+// Library/third-party includes
+
+
+namespace osvr {
+	namespace common {
+
+		namespace messages {
+			class EyeRegion::MessageSerialization {
+			public:
+				MessageSerialization(OSVR_EyeGazeDirection gaze,
+					//OSVR_EyeBufferElement *eyeBuff,
+					OSVR_ChannelCount sensor)
+					: m_data(gaze),
+					//m_eyeBuf(eyeBuff,
+					//[](OSVR_EyeBufferElement *) {}),
+					m_sensor(sensor) {}
+
+				//MessageSerialization() : m_eyeBuf(nullptr) {}
+				MessageSerialization() {}
+
+				/*
+				template <typename T>
+				void allocateBuffer(T &, size_t bytes, std::true_type const &) {
+					m_imgBuf.reset(reinterpret_cast<OSVR_EyeBufferElement *>(
+						malloc(bytes)),
+						&free);
+				}
+				*/
+				template <typename T>
```

Comment:
Can remove this function and the commented-out one above, as well as the commented out chunk of code in processMessage. They're for allocated a properly-aligned arbitrary-sized buffer for image data.